cellarium/cas/client.py

```diff
@@ -642,6 +794,69 @@ def search_10x_h5_file(
             feature_names_column_name=feature_names_column_name,
         )
 
+    def search_matrix(
+        self,
+        matrix: t.Union[str, "anndata.AnnData"],
+        chunk_size=CHUNK_SIZE_SEARCH_DEFAULT,
+        cas_model_name: str = "default",
+        count_matrix_input: constants.CountMatrixInput = constants.CountMatrixInput.X,
+        feature_ids_column_name: str = "index",
+        feature_names_column_name: t.Optional[str] = None,
+    ):
+        """
+        Send an instance of :class:`anndata.AnnData` to the Cellarium Cloud backend for nearest neighbor search. The
+        function splits the ``adata`` into smaller chunks and asynchronously sends them to the backend API service.
+        Each chunk is of equal size, except for the last one, which may be smaller. The backend processes
+        these chunks in parallel.
+
+        :param matrix: Either path to a file (must be either `.h5` or `.h5ad`) or an :class:`anndata.AnnData` instance
+            to annotate
+        :param chunk_size: Size of chunks to split on
+        :param cas_model_name: Model name to use for annotation. |br|
+            `Allowed Values:` Model name from the :attr:`allowed_models_list` list or ``"default"``
+            keyword, which refers to the default selected model in the Cellarium backend. |br|
+            `Default:` ``"default"``
+        :param count_matrix_input:  Where to obtain a feature expression count matrix from. |br|
+            `Allowed Values: Choices from enum :class:`cellarium.cas.constants.CountMatrixInput` |br|
+            `Default:` ``"CountMatrixInput.X"``
+        :param feature_ids_column_name: Column name where to obtain Ensembl feature ids. |br|
+            `Allowed Values:` A value from ``adata.var.columns`` or ``"index"`` keyword, which refers to index
+            column. |br|
+            `Default:` ``"index"``
+        :param feature_names_column_name: Column name where to obtain feature names (symbols).
+            feature names wouldn't be mapped if value is ``None`` |br|
+            `Allowed Values:` A value from ``adata.var.columns`` or ``"index"`` keyword, which refers to index
+            column. |br|
+            `Default:` ``None``
+
+        :return: A list of dictionaries with annotations for each of the cells from input adata
+        """
+        if isinstance(matrix, str):
+            matrix = _io.read_h5_or_h5ad(filename=matrix)
+
+        if chunk_size > 500:
```

Comment:
Can this be a constant?

Reply:
Meta point: if we made the input an Pydantic object, we could have validation that way no?

Reply:
Oh actually no, this one doesn't exist.:)

Reply:
> Meta point: if we made the input an Pydantic object, we could have validation that way no?

I think so. Might be a good idea, on the other hand, now the client API is basically a python call function, if it has request body as Pydantic model, it'd be not a very user friendly way. We can discuss it along with other schema related stuff we talked offline. Does that sound good? Looks like it is anyway out of the context for this PR.

Reply:
Yep, that's totally fair

Reply:
But having pydantic schemas as method return types for these methods instead of`t.List[t.Dict[str, t.Any]` would be reaallllllly helpful. And when we get to the point when we have shared schemas, it'd be just great to reuse those that are actually in Cellarium Cloud.

Reply:
great call!